File: Simplace/SimplaceETHargreaves/src/cs/SimplaceETHargreaves/ReferenceETHargreaves_Exogenous.cs
using System;
using System.Collections.Generic;

public class ReferenceETHargreaves_Exogenous 
{
    private double _iTMax;
    private double _iSolarRadiation;
    private double _iTMin;
    
    /// <summary>
    /// Constructor of the ReferenceETHargreaves_Exogenous component")
    /// </summary>  
    public ReferenceETHargreaves_Exogenous() { }
    
    
    public ReferenceETHargreaves_Exogenous(ReferenceETHargreaves_Exogenous toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            iTMax = toCopy.iTMax;
            iSolarRadiation = toCopy.iSolarRadiation;
            iTMin = toCopy.iTMin;
        }
    }
    public double iTMax
    {
        get { return this._iTMax; }
        set { this._iTMax= value; } 
    }
    public double iSolarRadiation
    {
        get { return this._iSolarRadiation; }
        set { this._iSolarRadiation= value; } 
    }
    public double iTMin
    {
        get { return this._iTMin; }
        set { this._iTMin= value; } 
    }
}

File: Simplace/SimplaceETHargreaves/src/java/SimplaceETHargreaves/ReferenceETHargreaves_Exogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class ReferenceETHargreaves_Exogenous
{
    private double iTMax;
    private double iSolarRadiation;
    private double iTMin;
    
    public ReferenceETHargreaves_Exogenous() { }
    
    public ReferenceETHargreaves_Exogenous(ReferenceETHargreaves_Exogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.iTMax = toCopy.getiTMax();
            this.iSolarRadiation = toCopy.getiSolarRadiation();
            this.iTMin = toCopy.getiTMin();
        }
    }
    public double getiTMax()
    { return iTMax; }

    public void setiTMax(double _iTMax)
    { this.iTMax= _iTMax; } 
    
    public double getiSolarRadiation()
    { return iSolarRadiation; }

    public void setiSolarRadiation(double _iSolarRadiation)
    { this.iSolarRadiation= _iSolarRadiation; } 
    
    public double getiTMin()
    { return iTMin; }

    public void setiTMin(double _iTMin)
    { this.iTMin= _iTMin; } 
    
}